documentation.lisp

```diff
@@ -685,6 +693,34 @@ See FIND-TEST
 See REMOVE-TEST
 See REFERENCED-DEPENDENCIES")
 
+  (variable *postprocessor-for-define-test+run*
+    "Contains a function which takes a list of four values as per
+DEFINE-TEST+RUN and, by default, returns all of them as values (with
+VALUES-LIST).  Its purpose is to easily let you customize the feedback from
+DEFINE-TEST+RUN to your own liking.")
```

Comment:
This seems too specific to include to me. A user could instead just define a new macro around `define-test+run` and use that instead to customise the results.

Reply:
Indeed. But the user-defined expansion would have to be smth like:

```
  (defmacro my-define-test+run (name &body body)
    `(progn (eval-when (:compile-toplevel :load-toplevel)
              (parachute:define-test ,name ,@body))
            (eval-when (:execute)
              (multiple-value-bind (name report status result)
                  (parachute:define-test+run ,name ,@body)
                ;; ...
                ))))
```

so the snippet would probably best be included in the docstring. With the function, it's just:

```
(setf parachute:*postprocessor-for-define-test+run*
  (lambda (v) (destructuring-bind (name report status result) v
                ...)))
```

Yet another way is to hide the variable and expose this instead:

```
  (defmacro def-result-for-define-test+run (&body body) ; untested
    `(setf *postprocessor-for-define-test+run*
       (lambda (x) (destructuring-bind (name report status result) x
                     (declare (ignorable name report status result))
                     ,@body))))
```

What do you think?

Reply:
Yes, the aesthetics. So, as of now, the return result for define-test+run looks like this:

```
=> SOME-TEST-0, #<PARACHUTE:PLAIN 7, PASSED results>, :PASSED, NIL
=> SOME-TEST-0, #<PARACHUTE:PLAIN 8, FAILED results>, :FAILED, (1 (IS EQL 0 1))
```

This is a bit noisy for me, so I prefer this:

```
=> #<PARACHUTE:PLAIN 7, PASSED results>
=> (:FAILED 1), (IS EQL 0 1)
```

// the 1 in `(:FAILED 1)` shows that only one test failed

I think that could be a decent default too, btw.

Reply:
Ok. I really don't want to include a mechanism like this just for aesthetics.

I'm not sure how to feel about variable return values, but I suppose having this just return something like
```
:FAILED, (IS EQL 0 1), (IS EQL 1 2), ...
:PASSED
```
Would be fine.

Reply:
Not a problem, please see the update. One thing: since there are implementation-defined limits on the number of multiple values (with the standard guaranteeing only 20) so I simply used a list for the failed clauses.

Reply:
Ah, good catch. That works for me!